proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala

```diff
@@ -517,35 +523,49 @@ object HttpApi {
         )
 
       if (methDesc.isServerStreaming) {
-        def toChunk(entityMessage: MessageOrBuilder): HttpEntity.Chunk =
-          HttpEntity.Chunk(
-            if (isHttpBodyResponse) extractDataFromHttpBody(entityMessage) ++ NEWLINE_BYTES
-            else ByteString(jsonPrinter.print(entityMessage)) ++ NEWLINE_BYTES
-          )
+        val sseAccepted = new MediaTypeNegotiator(grpcRequest.headers).isAccepted(MediaTypes.`text/event-stream`)
 
         data
           .prefixAndTail(1)
           .runWith(Sink.head)
-          .map {
+          .flatMap {
             case (Seq(protobuf: ProtobufAny), rest) =>
-              val entityMessage: MessageOrBuilder = parseResponseBody(protobuf)
-              val contentType =
-                if (isHttpBodyResponse) extractContentTypeFromHttpBody(entityMessage)
-                else ContentTypes.`application/json`
-
-              val first =
-                Source.single(toChunk(entityMessage))
-              val following =
-                rest.map(protobuf => toChunk(parseResponseBody(protobuf)))
-
-              //val extensions = extractExtensionsFromHttpBody(entityMessage) // FIXME / TODO HttpBody.extensions not supported yet
-
-              HttpResponse(
-                entity = HttpEntity.Chunked(
-                  contentType,
-                  first.concat(following)
+              if (isHttpBodyResponse) {
+                val entityMessage = parseResponseBody(protobuf) // We need to peek-ahead to check the content-type of the response we're sending out
+                Future.successful(
+                  HttpResponse(
+                    entity = HttpEntity.Chunked(
+                      extractContentTypeFromHttpBody(entityMessage),
+                      Source
+                        .single(entityMessage)
+                        .concat(rest.map(parseResponseBody))
+                        .map(em => HttpEntity.Chunk(extractDataFromHttpBody(em) ++ NEWLINE_BYTES))
+                    )
+                  )
+                )
+              } else if (sseAccepted) {
+                import EventStreamMarshalling._
+                Marshal(
+                  Source
+                    .single(protobuf)
+                    .concat(rest)
+                    .map(parseResponseBody)
+                    .map(em => ServerSentEvent(jsonPrinter.print(em)))
```

Comment:
Is sending gRPC responses as json SSE's a standard or something we're just implementing because it can be convenient?

Reply:
This is for the HTTP-API (REST)

Reply:
Right - is this SSE variation based on a published semi-standard like https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md or something we came up with ourselves?

Reply:
@raboof This is not grpc-web, it's an analogue of https://github.com/googleapis/googleapis/blob/master/google/api/http.proto (how GCP offers REST for their gRPC protocols)

Reply:
(we support grpc-web as well)

Reply:
Right, so it's a convention spearheaded by Google but not written down anywhere explicitly. :+1:

Reply:
@raboof in fact, sending a grpc stream via SSE was not invented anywhere, or not anywhere that I know of. This implementation here takes advantage of the HTTP API rest over gRPC (which is implemented by Google in GCP and some other proxy tools like [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway)) and adds support to the SSE at the top

Reply:
@sleipnir It was a great idea, Adriano! :)